### PR TITLE
Remove retry logging

### PR DESCRIPTION
**Pull request by wentinghu**

### CODE REVIEW

File: retry/retry.go [hunk 1]
```diff
@@ -82,7 +82,9 @@ func (r *Retrier) Retry(f func() (interface{}, error)) (interface{}, error) {
 		}
 
 		time.Sleep(next)
-		r.notifyRetry(err, numTries)
+		if len(r.notifyRetryFuncs) > 0 {
```
Comment (phobologic) [inline retry/retry.go]: I don't think you need this check - since if notifyRetryFuncs is a blank list, it just won't call anything.

Comment (sumeet) [inline retry/retry.go]: 👍 we don't need this

**Comment by sumeet** [comment]
👍

**Comment by sumeet** [comment]
code actually looks fine. i thought we were manually logging inside of hermes, but we're kinda depending on this in there.

i wonder why we're doing this? is this too noisy? this logging has actually be pretty useful inside r101-hermes - i think we may have to add it back in there manually

**Comment by phobologic** [comment]
@sumeet why do you think this logging is useful inside hermes?  I brought this up because when trying to troubleshoot issues in hermes all I saw were these log lines, which was basically just noise making finding actual problems a pain.

https://service.sumologic.com/ui/#section/search/NYKWgYJ7xnCHq9HMAC6Q54rwnQ5oULjg63DMWGWm

Do we do anything with these log lines?  Do we care every time we retry?

**Comment by sumeet** [comment]
ahhhh ok. so normally retrying should be business as usual, but i'm a little biased for this logging because i'm currently working on debugging an issue in hermes where retries are being exhausted and i'm trying to figure out why. another issue with that is that the current logging is insufficient and needs to be buffed in order to debug the issue.

so we're gonna have to customize the hermes logging anyway.

so 👍 to this change! don't wanna generate unnecessary noise in every service that happens to use it :)

**Comment by phobologic** [comment]
@sumeet keep in mind that there's a separate retry Logger attribute for when we've exhausted retries, just in case you thought there was just the one:

https://github.com/remind101/pkg/blob/master/retry/retry.go#L47

That isn't being changed, so you should still see when it fails.

**Comment by sumeet** [comment]
ah right!

**Comment by sumeet** [comment]
@ejholmes those aren't exported, but `AddNotifyRetry` and `AddNotifyGaveUp` are, so you could use those to set em. e.g., https://github.com/remind101/r101-hermes/blob/master/message_store.go#L530-L537

**Comment by ejholmes** [comment]
Ah! Missed those.

**Comment by phobologic** [comment]
@ejholmes So do you think we should remove the default notify's for GaveUp as well?  ANyway we can figure out who uses `retry` and see if we should setup a notify specifically for them if we remove it.

**Comment by ejholmes** [comment]
> So do you think we should remove the default notify's for GaveUp as well

If we're already moving logging elsewhere, I think this makes sense.

> ANyway we can figure out who uses retry and see if we should setup a notify specifically for them if we remove it.

I think probably the best way to make this change, without breaking existing consumers (breaking is a strong word, since it's just logging), would be to just add a `New` factory method that returns a Retryier without any of the logging added.

**Comment by wentinghu** [comment]
@ejholmes So make a NewNewRetrier func which is a Retrier with empty arrays for notifyRetryFuncs, notifyGaveUpFuncs, and notifyShouldNotRetry? :P

**Comment by ejholmes** [comment]
@wentinghu IMO, just call it `New` (e.g. `retry.New`).

File: retry/retry.go [hunk 2]
```diff
@@ -48,6 +48,18 @@ func NewRetrier(name string,
 		notifyShouldNotRetryFuncs: []RetryNotifier{logShouldNotRetry}}
 }
 
+func New(name string,
+	backOffOpts *BackOffOpts, shouldRetryFunc func(error) bool) *Retrier {
+
+	return &Retrier{
+		Name:                      fmt.Sprintf("%s%d", name, atomic.AddUint32(&retrierNum, 1)),
```
Comment (ejholmes) [inline retry/retry.go]: Minor, but I would move this into a `func newName` since it's now duplicated in some places.

**Pull request merged**